File: backend/app/models/model_config.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String, Boolean
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base
from typing import Optional


class ModelConfig(Base):
    __tablename__ = "model_configs"

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    display_name: Mapped[str] = mapped_column(String(100), nullable=False)
    provider: Mapped[str] = mapped_column(String(20), nullable=False)
    mode: Mapped[str] = mapped_column(String(20), nullable=False)
    api_format: Mapped[str] = mapped_column(String(50), nullable=False, default="openai_chat")
    api_key: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
    base_url: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
    full_url: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    model: Mapped[str] = mapped_column(String(2000), nullable=False)
    multimodal: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    model_series: Mapped[str] = mapped_column(String(50), nullable=False, default="default")
    context_input: Mapped[int] = mapped_column(Integer, default=128000, nullable=False)
    context_output: Mapped[int] = mapped_column(Integer, default=4096, nullable=False)
    tool_call_rounds: Mapped[int] = mapped_column(Integer, default=200, nullable=False)
    enabled: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    monthly_quota: Mapped[int] = mapped_column(Integer, default=1000000, nullable=False)
    used_tokens: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.now, nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.now, onupdate=datetime.now, nullable=False
    )

File: backend-go/models/model_config.go
package models

import (
	"time"
)

type ModelConfig struct {
	ID             string    `orm:"column(id);pk;size(64)" json:"id"`
	Name           string    `orm:"column(name);size(100)" json:"name"`
	DisplayName    string    `orm:"column(display_name);size(100)" json:"display_name"`
	Provider       string    `orm:"column(provider);size(20)" json:"provider"`
	Mode           string    `orm:"column(mode);size(20)" json:"mode"`
	APIFormat      string    `orm:"column(api_format);size(50);default(openai_chat)" json:"api_format"`
	APIKey         string    `orm:"column(api_key);size(500);null" json:"api_key"`
	BaseURL        string    `orm:"column(base_url);size(500);null" json:"base_url"`
	FullURL        bool      `orm:"column(full_url);default(false)" json:"full_url"`
	Model          string    `orm:"column(model);size(2000)" json:"model"`
	Multimodal     bool      `orm:"column(multimodal);default(false)" json:"multimodal"`
	ModelSeries    string    `orm:"column(model_series);size(50);default(default)" json:"model_series"`
	ContextInput   int       `orm:"column(context_input);default(128000)" json:"context_input"`
	ContextOutput  int       `orm:"column(context_output);default(4096)" json:"context_output"`
	ToolCallRounds int       `orm:"column(tool_call_rounds);default(200)" json:"tool_call_rounds"`
	Enabled        bool      `orm:"column(enabled);default(false)" json:"enabled"`
	MonthlyQuota   int       `orm:"column(monthly_quota);default(1000000)" json:"monthly_quota"`
	UsedTokens     int       `orm:"column(used_tokens);default(0)" json:"used_tokens"`
	SortOrder      int       `orm:"column(sort_order);default(0)" json:"sort_order"`
	CreatedAt      time.Time `orm:"column(created_at);auto_now_add;type(datetime)" json:"created_at"`
	UpdatedAt      time.Time `orm:"column(updated_at);auto_now;type(datetime)" json:"updated_at"`
}

func (m *ModelConfig) TableName() string {
	return "model_configs"
}
